### DISCUSSION TITLE
Make sure you use nightly toolchain to run formatter

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1108] [General]
We now have rustfmt configured to run on a few of the directories in `rust-bitcoin`. If you are getting unexplained formatting changes in your commits be sure to check that you are using the nightly toolchain when formatting (incl. your editor and format-on-save).

If there are any other teething problems please mention them or poke me if you want something fixed.

Thanks.

**Comment by dpc** [comment]
Another benefit of accommodatin Nix flakes in project setup - you can lock reference versions of all tools, including `rustfmt` and get consistent results across all developers and environments.

I need to finally write down a longer blog post about it, but in RustShop [the CI](https://github.com/rustshop/rustshop/blob/89a40217d05a04f777e59239992450c1ad76d572/.github/workflows/ci-services.yml#L36), [auto-set github hooks](https://github.com/rustshop/rustshop/blob/89a40217d05a04f777e59239992450c1ad76d572/flake.nix#L185), etc all automatically [enforce the same formatting](https://github.com/rustshop/rustshop/blob/89a40217d05a04f777e59239992450c1ad76d572/flake.nix#L191) at all stages of development. With a bit of caching, [you can get linting results in only 40s after the CI started](https://github.com/rustshop/rustshop/runs/7375374070?check_suite_focus=true).

**Reply by tcharding** [reply]
Cheers man, to be honest I did not dig too deeply into RustShop because I personally hate devops. I do just enough of it to make my dev life better, and hopefully that of others, but I do not find it to be engaging work :)

**Reply by tcharding** [reply]
I'll definitely review PRs and try help out if you want to change the way we do things if it means we get closer to a place were devs can dev and all the devops  stuff is seamless :)

**Reply by dpc** [reply]
Please just ignore the `devops` part. RustShop is supposed to be just "whole software shop", and a very large part of that is devops - AWS, k8s, etc. I dig that stuff a lot. However the dev stuff there is completely separate, and it's not necessary to touch anything devops to use it. In essence it's just "Rust + Nix for everything".

I understand that Nix is yet another thing with a steep learning curve, but IMO Nix flakes are phenomenal tool, and when embraced can replaced and take to another level the whole dev tooling. A Nix Flake is like `Cargo.lock` *for everything* - bringing it in a reproducible, pure (in a FP sense) way.

**Reply by tcharding** [reply]
ok cool, I'll put some more time into it. Thanks man.
